src/modules/Game.class.js

```diff
@@ -50,19 +138,168 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.#status;
+  }
+
+  /**
+   * Update game status.
+   *
+   * @param {string} value
+   *
+   */
+  setStatus(value) {
+    this.#status = value;
+  }
+
+  /**
+   * This method shoud be private, but the tests consider it as public
+   */
+  start() {
+    this.#score = 0;
+    this.#status = 'playing';
+    this.#state = this.#getStateCopy(this.#initialState);
+
+    // Is called twice by requirements
+    this.#randomizer();
+    this.#randomizer();
+  }
+
+  // Resets the game.
+  restart() {
+    this.start();
+    this.#status = 'idle';
+  }
 
   /**
-   * Starts the game.
+   * Put 2 or 4 in freely cells if they exists 
    */
-  start() {}
+  #randomizer() {
+    const empties = [];
+    let idxSorted;
+
+    for (let line = 0; line < 4; line++) {
+      for (let column = 0; column < 4; column++) {
+        if (this.#state[line][column] === 0) {
+          empties.push([line, column]);
+        }
+      }
+    }
+
+    if (empties.length > 0) {
+      idxSorted = Math.floor(Math.random() * empties.length);
+
+      this.#state[empties[idxSorted][0]][empties[idxSorted][1]] =
+        Math.floor(Math.random() * 10) < 9 ? 2 : 4;
+    }
+  }
+
+  // Matrix transformation transpose operation.
+  #tranpose() {
+    let tempRetriveVar;
+
+    for (let ln = 0; ln < 4; ln++) {
+      for (let cl = ln; cl < 4; cl++) {
+        tempRetriveVar = this.#state[ln][cl];
+        this.#state[ln][cl] = this.#state[cl][ln];
+        this.#state[cl][ln] = tempRetriveVar;
+      }
+    }
+  }
+
+  // Revert each line of this.#state, dont use to revert any array. 
+  #reverse() {
+    this.#state.forEach((e, i, arr) => arr[i].reverse());
+  }
+
+  // Check if movements isn't possible than change this.status.
+  #checkIfLose() {
+    let allSides = [];
+
+    for (let column = 0; column < this.#SIZE; column++) {
+      for (let line = 0; line < this.#SIZE; line++) {
+        if (this.#state[line][column] !== 0) {
```

Comment:
What if it `this.#state[line][column]` is 0; what does it mean? Use early returns whenever possible to avoid unnecessary processes